### PR TITLE
Add support with async/await using Promises. 

### PR DESCRIPTION
**Pull request by lilezek**
Some callbacks now are more restrictive.

### CODE REVIEW

File: src/service/index.ts [hunk 1]
```diff
@@ -52,6 +57,21 @@ export class Service {
...
 
+  /**
+   * Subscribe to a topic, getting a producer/consumer array of promises.
+   * @param {string} topic
+   * @param {boolean} disableGroup
+   * @returns An async array following the producer/consumer pattern.
+   */
+  public onAsync<T = any>(topic: string, disableGroup: boolean = false) {
+    const SUBJECT = `${this.name}:${topic}`;
+    const array = new AsyncArray<T>();
```
Comment (ivaylopivanov) [inline src/service/index.ts]: Please change to upper case

File: src/service/index.ts [hunk 2]
```diff
@@ -216,7 +267,7 @@ export class Service {
     }
 
     if (!route) {
-      return callback(new Error('Invalid request path'));
+      return callback(new Response({}, new OrionError('Invalid request path')));
```
Comment (ivaylopivanov) [inline src/service/index.ts]: better `return callback(new Response(null, new OrionError('Invalid request path')));`

File: src/service/index.ts [hunk 3]
```diff
@@ -233,14 +284,15 @@ export class Service {
...
     if (res instanceof Response === false) {
       throw new Error('The response must instance of Orion.Response');
     }
     if (res.error !== undefined &&
         res.error instanceof OrionError === false) {
       throw new Error('The passed error must instance of Orion.Error');
     }
-  }
 
+    return true;
```
Comment (ivaylopivanov) [inline src/service/index.ts]: why do we need to return true?

Comment (lilezek) [reply src/service/index.ts]: Is a user defined type guard requirement.

**Pull request merged**